Dockerfile

```diff
@@ -2,8 +2,8 @@ FROM node:5
 
 RUN apt-get install libpq-dev
 
-COPY package.json /
+WORKDIR /sequelize
+COPY package.json /sequelize/
 RUN npm install
 
-WORKDIR /sequelize
-VOLUME /sequelize
+COPY . /sequelize/
```

Comment:
You can try to create different volumes for each folder but it doesn't look very efficient (you'll probably also need the `.jshintrc` file for the lint test, I don't know about the rest of the files).
A copy operation is actually very fast so rebuilding the container only takes a few seconds. But it's really your choice, I just thought to give a hand as I had some experience with docker containers in the past and it didn't look like the dockerfile was doing what you expected.